api/osf_api.py

```diff
@@ -1050,3 +1050,125 @@ def get_project_node_analytics_data(session, node_id=None, timespan='week'):
     url = '_/metrics/query/node_analytics/{}/{}/'.format(node_id, timespan)
     data = session.get(url)['data']
     return data or None
+
+
+def get_fake_file_guid(session, file_id):
+
+    url = '/v2/files' + file_id
+    data = session.get(url=url)
+    file_guid = data['data']['attributes']['guid']
+    return file_guid
+
+
+def get_existing_file_data(session, node_id=settings.PREFERRED_NODE):
+    """Return the id of the first file in OSFStorage on a given node.
+    Uploads a new file if one does not exist.
+    """
+    node = client.Node(session=session, id=node_id)
+    node.get()
+    files_url = node.relationships.files['links']['related']['href']
+    data = session.get(files_url + 'osfstorage/')
+    name = data['data'][0]['attributes']['name']
+    return name, data
+
+
+def update_custom_project_metadata(session, node_id):
+    """Updates project metadata fields resource_type and
+    resource_language and support funder info with custom values"""
+    url = 'v2/custom_item_metadata_records/{}/'.format(node_id)
+    raw_payload = {
+        'data': {
+            'id': node_id,
+            'type': 'custom-item-metadata-records',
+            'attributes': {
+                'language': 'eng',
+                'resource_type_general': 'Collection',
+            },
+        }
+    }
+    session.put(url=url, raw_body=json.dumps(raw_payload))
+
+
+def delete_project_contributor(session, node_id, user_name):
```

Comment:
Not a request but a comment: if contributor is the only admin, this api request will fail. I don't think it affects what we are doing here in the tests.